e2e/file/promote_test.go

```diff
@@ -0,0 +1,112 @@
+package file
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"sigs.k8s.io/k8s-container-image-promoter/pkg/cmd"
+	"sigs.k8s.io/yaml"
+)
+
+const ProdBucket = "gs://k8s-cip-test-prod"
+const StagingBucket = "gs://k8s-staging-cip-test"
+
+func TestPromote(t *testing.T) {
+	for _, bucket := range []string{StagingBucket, ProdBucket} {
+		files, err := ListFiles(bucket)
+		if err != nil {
+			t.Fatalf("failed to dump bucket %s: %v", bucket, err)
+		}
+
+		if err := DeleteAllFiles(files); err != nil {
+			t.Fatalf("failed to empty bucket %s: %v", bucket, err)
+		}
+	}
+
+	// We upload a file into the prod bucket, outside of the subdirectory - it should be untouched
+	EnsureFile(t, ProdBucket+"/index.html", []byte("<html><body>k8s-cip-test-prod</body></html>"))
+
+	for i := 1; i < 100; i += 10 {
```

Comment:
Can you either comment on the purpose of this for-loop or move it out to a separate function?